addons/common/events/predictions/src/main/java/org/kie/kogito/eventdriven/predictions/EventDrivenPredictionsController.java

```
@@ -0,0 +1,232 @@
+        return Optional.ofNullable(event).filter(e -> REQUEST_EVENT_TYPE.equals(e.getType()));
+    }
+
+    private Optional<EvaluationContext> buildEvaluationContext(CloudEvent event) {
+        KogitoExtension kogitoExtension = ExtensionProvider.getInstance().parseExtension(KogitoExtension.class, event);
+        Map<String, Object> data = CloudEventUtils.decodeMapData(event, String.class, Object.class).orElse(null);
+
+        if (kogitoExtension == null) {
+            LOG.warn("Received CloudEvent(id={} source={} type={}) with null Kogito extension", event.getId(), event.getSource(), event.getType());
+        }
+
+        if (data == null) {
+            LOG.warn("Received CloudEvent(id={} source={} type={}) with null data", event.getId(), event.getSource(), event.getType());
+        }
+
+        return Optional.of(new EvaluationContext(event, kogitoExtension, data));
+    }
+
+    private EvaluationContext processRequest(EvaluationContext ctx) {
+        if (!ctx.isValidRequest()) {
+            ctx.setResponseError(PredictionsResponseError.BAD_REQUEST);
+            return ctx;
+        }
+
+        Optional<PredictionModel> optPredictionModel = getPredictionModel(ctx.getRequestModelName());
+        if (!optPredictionModel.isPresent()) {
+            ctx.setResponseError(PredictionsResponseError.MODEL_NOT_FOUND);
+            return ctx;
+        }
+
+        PredictionModel model = optPredictionModel.get();
+        PMMLContext context = model.newContext(ctx.getRequestData());
+        PMML4Result apiResult = model.evaluateAll(context);
+
+        ctx.setResult(apiResult);
+        return ctx;
+    }
+
+    private Optional<PredictionModel> getPredictionModel(String modelName) {
+        try {
+            return Optional.ofNullable(predictionModels.getPredictionModel(modelName));
+        } catch (RuntimeException e) {
+            LOG.warn("Model not found with name=\"{}\"", modelName);
```

@kostola 
I'm not sure of that: what could lead to an "ignorable" runtime exception here ?
Said differently: if 
`predictionModels.getPredictionModel(modelName))` throws an exception, there may be an issue all over the code pasth that lead to `PMMLRuntime` instantiation, and I think we should not simply hide that...

---

I am catching a `RuntimeException` because that's what `getPredictionModel` throws when a model is not found.
I will introduce a specific exception to catch so that all the others (caused by unexpected issues) will be propagated as it should be.

---

great, thanks!

---

Beside that, that exception (whatever it is) should not be swallowed away: if the user ask evaluation of a model that is not available, an error or similar should be raised

---

I added `PredictionModelNotFoundException` for this specific case